searchlib/src/vespa/searchlib/common/rcuvector.hpp

```diff
@@ -54,8 +54,9 @@ void
 RcuVectorBase<T>::expand(size_t newCapacity) {
     std::unique_ptr<Array> tmpData(new Array());
     tmpData->reserve(newCapacity);
-    tmpData->resize(_data.size());
-    memcpy(tmpData->begin(), _data.begin(), _data.size() * sizeof(T));
+    for (const T & v : _data) {
+        tmpData->push_back_fast(v);
+    }
     tmpData->swap(_data); // atomic switch of underlying data
```

Comment:
Unrelated to this commit, but: the comment says "atomic switch of underlying data", but `vespalib::Array::swap()` is comprised of two separate variable swaps, neither of which being atomic by themselves or together. Just wondering if the comment is outdated or if this happens in a context where concurrent reading of `_data` can't happen.

Reply:
That is correct. The term atomic is not correct, nor necessary. This happens under a single writer, and the readers are guaranteed to read data that are valid until their generation is released. If I am not mistaken. It is a pointer and a size variable. Since size will be identical in both a and b and only the pointers change you should be fine.